File: frontend/src/model/model/entity/SiteAuthor.ts
import BaseEntity from '../base/BaseEntity.ts'
import { notNullish } from '@renderer/utils/CommonUtil.ts'
import BaseAuthor from '../interface/BaseAuthor.ts'

/**
 * 站点作者
 */
export default class SiteAuthor extends BaseEntity implements BaseAuthor {
  /**
   * 作者来源站点id
   */
  siteId: number | undefined | null
  /**
   * 站点中作者的id
   */
  siteAuthorId: string | undefined | null
  /**
   * 站点中作者的名称
   */
  authorName: string | undefined | null
  /**
   * 站点中作者的固定名称（无法更改的，比如账号）
   */
  fixedAuthorName: string | undefined | null
  /**
   * 站点中作者的曾用名
   */
  siteAuthorNameBefore: string[] | string | undefined | null
  /**
   * 介绍
   */
  introduce: string | undefined | null
  /**
   * 站点作者在本地对应的作者id
   */
  localAuthorId: number | undefined | null
  /**
   * 最后一次使用的时间
   */
  lastUse: number | null | undefined

  constructor(siteAuthor?: SiteAuthor) {
    super(siteAuthor)
    if (notNullish(siteAuthor)) {
      this.siteId = siteAuthor.siteId
      this.siteAuthorId = siteAuthor.siteAuthorId
      this.authorName = siteAuthor.authorName
      this.fixedAuthorName = siteAuthor.fixedAuthorName
      if (typeof siteAuthor.siteAuthorNameBefore === 'string') {
        this.siteAuthorNameBefore = siteAuthor.siteAuthorNameBefore.split(',')
      } else {
        this.siteAuthorNameBefore = siteAuthor.siteAuthorNameBefore
      }
      this.introduce = siteAuthor.introduce
      this.localAuthorId = siteAuthor.localAuthorId
      this.lastUse = siteAuthor.lastUse
    }
  }
}

File: backend/base/model/entity/site_author.go
package entity

import (
	"database/sql"

	"github.com/library-squirrel/backend/base/model"
)

// SiteAuthor 站点作者
type SiteAuthor struct {
	*model.BaseEntity
	SiteID               sql.NullInt64  `gorm:"column:site_id;uniqueIndex:idx_site_author_site_site_author" json:"siteId"`
	SiteAuthorID         sql.NullString `gorm:"column:site_author_id;uniqueIndex:idx_site_author_site_site_author" json:"siteAuthorId"`
	AuthorName           sql.NullString `gorm:"column:author_name" json:"authorName"`
	FixedAuthorName      sql.NullString `gorm:"column:fixed_author_name" json:"fixedAuthorName"`
	SiteAuthorNameBefore sql.NullString `gorm:"column:site_author_name_before" json:"siteAuthorNameBefore"`
	Introduce            sql.NullString `gorm:"column:introduce" json:"introduce"`
	Homepage             sql.NullString `gorm:"column:homepage" json:"homepage"`
	LocalAuthorID        sql.NullInt64  `gorm:"column:local_author_id" json:"localAuthorId"`
	LastUse              sql.NullInt64  `gorm:"column:last_use" json:"lastUse"`
	// AvatarStoreID 头像文件的 persistent_store 行内嵌引用（作者行发起入库，拉取/导入编排事务内写；
	// 重拉与元数据回写均不进 upsert 覆盖域，随行保留）
	AvatarStoreID sql.NullInt64 `gorm:"column:avatar_store_id" json:"avatarStoreId"`
	// AvatarSourceURL 拉取期插件回报的头像来源 URL（站点声明权威域，进 upsert 覆盖域；
	// 后续拉取以其为变更检测比对键）
	AvatarSourceURL sql.NullString `gorm:"column:avatar_source_url" json:"avatarSourceUrl"`
}

func NewSiteAuthor() *SiteAuthor {
	return &SiteAuthor{
		BaseEntity: &model.BaseEntity{},
	}
}

func (SiteAuthor) TableName() string {
	return "site_author"
}
